internal/examples/mock/mock.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mock
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/googleapis/gax-go/v2"
+	translatepb "google.golang.org/genproto/googleapis/cloud/translate/v3"
+)
+
+// TranslationClient is used to translate text.
+type TranslationClient interface {
+	TranslateText(ctx context.Context, req *translatepb.TranslateTextRequest, opts ...gax.CallOption) (*translatepb.TranslateTextResponse, error)
+}
+
+// TranslateTextWithAbstractClient translates text to the targetLand using the
+// provided client.
+func TranslateTextWithAbstractClient(client TranslationClient, text string, targetLang string) (string, error) {
```

Comment:
Perhaps `InterfaceClient` since abstract isn't a thing in Go?

Reply:
Done